### PR TITLE
mgr/dashboard: Update to Angular 9

### PR DESCRIPTION
**Pull request by tspmelo**
This includes the automatic update using `ng update`, some manual changes suggested by the angular team, fixes to problems and manual update to all packages.

This should only be ready/merged after octopus is released.

More information regarding Angular 9: https://blog.angular.io/version-9-of-angular-now-available-project-ivy-has-arrived-23c97b63cfa3

## Checklist
- [x] https://github.com/ceph/ceph/pull/34442
- [x] #34285

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by Devp00l** [comment]
Could you add to the title and the description that this PR should be merged after your Cypress PR has been merged?

**Comment by Devp00l** [comment]
Just tested compared the unit test speed difference with or without our dev setting on it's about 100s of improvement. This also means that our initialization hack is still a lot faster than the improvements the angular team has done, hope they catch up sometime ;) in behalf of all other users.

Here are my results

| Description        | DEV == true           | DEV == False  |
| ------------- |:-------------:| -----:|
| ceph/master      | 347s | 1456s |
| tspmelo/wip-angular-9      | 254s      |  1344s |
| Difference | 93s (~27%)   |   112s (~8%)  |

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
I have removed a couple of commits that are not essential for this to be merged.
Will create a PR for each later.

File: src/pybind/mgr/dashboard/frontend/cd.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    logger(`Preparing build of EN.`);
+  } else {
+    langs = langs.split(/[ ,]/);
+    logger(`Preparing build of EN and ${langs}.`);
+  }
+
+  let locales = {};
+
+  langs.forEach((lang) => {
+    switch (lang) {
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/cd.js:70]: How about don't have so a long switch statement?

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/cd.js b/src/pybind/mgr/dashboard/frontend/cd.js
index 0887347201b..a2015de0b28 100755
--- a/src/pybind/mgr/dashboard/frontend/cd.js
+++ b/src/pybind/mgr/dashboard/frontend/cd.js
@@ -66,70 +66,38 @@ function prepare_build_2() {
 
   let locales = {};
 
+  const langSwitch = {
+    'cs': 'cs',
+    'de': 'de-DE',
+    'es': 'es-ES',
+    'fr': 'fr-FR',
+    'id': 'id-ID',
+    'it': 'it-IT',
+    'ja': 'ja-JP',
+    'ko': 'ko-KR',
+    'pl': 'pl-PL',
+    'pt': 'pt-BR'
+  }
+
   langs.forEach((lang) => {
-    switch (lang) {
-      case 'cs':
-      case 'cs-US':
-        locales['cs'] = { translation: 'src/locale/messages.cs.xlf', baseHref: '' };
-        break;
-
-      case 'de':
-      case 'de-DE':
-        locales['de'] = { translation: 'src/locale/messages.de-DE.xlf', baseHref: '' };
-        break;
-
-      case 'es':
-      case 'es-ES':
-        locales['es'] = { translation: 'src/locale/messages.es-ES.xlf', baseHref: '' };
-        break;
-
-      case 'fr':
-      case 'fr-FR':
-        locales['fr'] = { translation: 'src/locale/messages.fr-FR.xlf', baseHref: '' };
-        break;
-
-      case 'id':
-      case 'id-ID':
-        locales['id'] = { translation: 'src/locale/messages.id-ID.xlf', baseHref: '' };
-        break;
-
-      case 'it':
-      case 'it-IT':
-        locales['it'] = { translation: 'src/locale/messages.it-IT.xlf', baseHref: '' };
-        break;
-
-      case 'ja':
-      case 'ja-JP':
-        locales['ja'] = { translation: 'src/locale/messages.ja-JP.xlf', baseHref: '' };
-        break;
-
-      case 'ko':
-      case 'ko-KR':
-        locales['ko'] = { translation: 'src/locale/messages.ko-KR.xlf', baseHref: '' };
-        break;
-
-      case 'pl':
-      case 'pl-PL':
-        locales['pl'] = { translation: 'src/locale/messages.pl-PL.xlf', baseHref: '' };
-        break;
-
-      case 'pt':
-      case 'pt-BR':
-        locales['pt'] = { translation: 'src/locale/messages.pt-BR.xlf', baseHref: '' };
-        break;
-
-      case 'zh-Hans':
-      case 'zh-CN':
-        locales['zh-Hans'] = { translation: 'src/locale/messages.zh-CN.xlf', baseHref: '' };
-        break;
-
-      case 'zh-TW':
-      case 'zh-Hant':
-        locales['zh-Hant'] = { translation: 'src/locale/messages.zh-TW.xlf', baseHref: '' };
-        break;
-
-      default:
-        break;
+    const short = lang.substring(0,2)
+    if (short !== 'zh') {
+      const langSuffix = langSwitch[short]
+      if (langSuffix) {
+        locales[short] = { translation: `src/locale/messages.${langSuffix}.xlf`, baseHref: '' };
+      }
+    } else {
+      switch (lang) {
+        case 'zh-Hans':
+        case 'zh-CN':
+          locales['zh-Hans'] = {translation: 'src/locale/messages.zh-CN.xlf', baseHref: ''};
+          break;
+
+        case 'zh-TW':
+        case 'zh-Hant':
+          locales['zh-Hant'] = {translation: 'src/locale/messages.zh-TW.xlf', baseHref: ''};
+          break;
+      }
     }
   });
 
```

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
@Devp00l  Fixed all comments.

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/cd.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    langs = [];
+    logger(`Preparing build of EN.`);
+  } else {
+    langs = langs.split(/[ ,]/);
+    logger(`Preparing build of EN and ${langs}.`);
+  }
+
+  let angular = require(filename);
+
+  let allLocales = angular['projects']['ceph-dashboard']['i18n']['locales'];
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/cd.js:62]: Nice improvement :+1:

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by callithea** [comment]
- [pkg build](https://shaman.ceph.com/builds/ceph/wip-laura-testing-33121/) [WIP]
- QA run [PENDING]

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by callithea** [comment]
Re-scheduled the QA run with the latest changes of this PR (also wanted to check if the previous errors are reproducible)
- [pkg build](https://shaman.ceph.com/builds/ceph/wip-laura-testing-33121-2/) [WIP]
- QA run [PENDING]

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Pull request merged**